java/src/com/ibm/streamsx/topology/generator/spl/SPLGenerator.java

```diff
@@ -83,6 +98,406 @@ public String generateSPL(JsonObject graph) throws IOException {
+        parallelInfo.add("partitionedPorts", partitionedPorts);
+        
+        for(JsonObject startOp : startsEndsAndOperators.get(0)){
+            if(startOp.has("config") && startOp.get("config").getAsJsonObject().has(OpProperties.WIDTH)){
+                JsonElement width = startOp.get("config").getAsJsonObject().get(OpProperties.WIDTH);
+                parallelInfo.add(OpProperties.WIDTH, width);
+            }
+            
+            // If it's a physical source start operator, ignore its output port, it has no partition information.
+            if(startOp.has("config") && (hasAny(object(startOp, "config"), compOperatorStarts)))
+                continue;
+            
+            // Otherwise, get the partition information.
+            JsonObject outputPort = array(startOp, "outputs").get(0).getAsJsonObject();
+            JsonObject inputPort = array(startOp, "inputs").get(0).getAsJsonObject();
+            
+            // Set the width if it was contained in the output port.
+            if(outputPort.has(OpProperties.WIDTH))
+                parallelInfo.add(OpProperties.WIDTH, outputPort.get(OpProperties.WIDTH));
+            
+            
+            if(jstring(outputPort, PortProperties.ROUTING).equals(Routing.BROADCAST.toString())){
+                broadcastPorts.add(inputPort.get("name"));
+            }
+            
+            if(outputPort.has(PortProperties.PARTITIONED) && jboolean(outputPort, PortProperties.PARTITIONED)){
+                JsonObject partitionInfo = new JsonObject();
+                partitionInfo.add("name", inputPort.get("name"));
+                partitionInfo.add(PortProperties.PARTITION_KEYS, outputPort.get(PortProperties.PARTITION_KEYS));
+                partitionedPorts.add(partitionInfo);
+                
+                // There is at least one partitioned port, therefore it is partitioned.
+                compositeInvocation.addProperty("partitioned", true);
+            }      
+                
+        } 
+        
+        compositeInvocation.add("parallelInfo", parallelInfo);
+        compositeInvocation.addProperty("parallelOperator", true);
+        return compositeInvocation;
+    }
+
+    private JsonObject createLowLatencyCompositeInvocation(JsonObject opDefinition, List<List<JsonObject>> startsEndsAndOperators) {
+        JsonObject compositeInvocation = createCompositeInvocation(opDefinition, startsEndsAndOperators);
+        
+        return compositeInvocation;
+    }
+
+    private List<List<JsonObject> > findCompositeOpsOfAType(JsonObject graph, String startKind, String endKind, String opStartParam){
+        
+        for(JsonElement jePotentialStart : graph.getAsJsonArray("operators")){
+            JsonObject potentialStart = jePotentialStart.getAsJsonObject();
+            
+            // We've found a potential start to a composite. See if the composite doesn't contain another composite.   
+            if(kind(potentialStart).equals(startKind) || 
```

Comment:
Need comments as to why a config section makes an operator a potential start.

Especially if most operators are going to have config sections.

Should instead a potential start be identified by a function and that what gets evaluated against the operator. Then parallel would have something like:
`kind == $Parallel$ of op config has parallel info`
while low latency would have
`kind == $LowLatency`

Reply:
It's not just having a `config` section. It's both having a `config` section and the `config` containing the parameter indicating it is a `start`.

Since not all operators have configs, we first need to check if the config exists before accessing it in order to avoid a NPE.

```
op.has("config") && jboolean(object(op, "config"), opStartParam)
```

I added two functions: one which perform the logic of determining whether an operator is a (i.e., any) start op, and another which determines whether the operator is a start op for a particular type of region.